server/service/service_osquery.go

```diff
@@ -517,12 +517,15 @@ func (svc *Service) GetDistributedQueries(ctx context.Context) (map[string]strin
 		queries[hostLabelQueryPrefix+name] = query
 	}
 
-	liveQueries, err := svc.liveQueryStore.QueriesForHost(host.ID)
-	if err != nil {
-		return nil, 0, osqueryError{message: "retrieve live queries: " + err.Error()}
-	}
-	for name, query := range liveQueries {
-		queries[hostDistributedQueryPrefix+name] = query
+	if liveQueries, err := svc.liveQueryStore.QueriesForHost(host.ID); err != nil {
+		// If the live query store fails to fetch queries we still want the hosts
+		// to receive all the other queries (details, policies, labels, etc.),
+		// thus we just log the error.
+		level.Error(svc.logger).Log("op", "QueriesForHost", "err", err)
+	} else {
+		for name, query := range liveQueries {
+			queries[hostDistributedQueryPrefix+name] = query
+		}
 	}
```

Comment:
Just a thought: A possible optimization (for another time/PR) is: If there are live queries available for the host then return those only to the hosts (this would make live querying a bit faster as the hosts just have to execute the selected live queries instead of all policies/labels/details/etc.).

Reply:
This could be useful. Want to file an issue? We would have to be careful that we didn't end up in a scenario where Fleet keeps thinking the host needs to execute the same live query and then never sends the rest of the queries that need to be run as distributed queries.

Reply:
#3576